packages/forms/src/directives/ng_model.ts

```diff
@@ -233,6 +242,36 @@ export class NgModel extends NgControl implements OnChanges, OnDestroy {
           this.formDirective.removeControl({name: oldName, path: this._getPath(oldName)});
         }
       }
+      if (ngDevMode && this.formDirective) {
```

Comment:
I'd propose moving the logic to a separate function for readability (to keep the `ngOnChanges` as simple as possible):
```suggestion
      if (ngDevMode) {
        _checkDuplicateModels(this, changes);
      }
```
Note: the `_checkDuplicateModels` is a standalone function (**not** within the NgModel class), so that it can be fully tree-shaken away.

Reply:
Moved to standalone, took form check with it.